### PR TITLE
feat(cli): initialize inline environments

### PR DESCRIPTION
**Pull request by Duologic**
initialize inline environments with `tk env add --inline` and `tk init --inline`

### CODE REVIEW

File: cmd/tk/util.go [hunk 1]
```diff
@@ -56,3 +58,22 @@ func writeJSON(i interface{}, path string) error {
 
 	return nil
 }
+
+// writeJSON writes the given object to the path as a JSON file
```
Comment (jvrplmlmn) [inline cmd/tk/util.go:62]: Comment and function signature don't match.

File: cmd/tk/util.go [hunk 2]
```diff
@@ -56,3 +58,22 @@ func writeJSON(i interface{}, path string) error {
...
+func writeJsonnet(i interface{}, path string) error {
+	out, err := json.MarshalIndent(i, "", "  ")
+	if err != nil {
+		return fmt.Errorf("marshalling: %s", err)
+	}
+
+	main, err := formatter.Format(path, string(out), formatter.DefaultOptions())
+	if err != nil {
+		return fmt.Errorf("writing %s: %s", path, err)
```
Comment (jvrplmlmn) [inline cmd/tk/util.go:71]: `formatter.Format` does not write the file, I have added a suggestion to update the error message:

```suggestion
		return fmt.Errorf("formatting %s: %s", path, err)
```

File: cmd/tk/env.go [hunk 3]
```diff
@@ -164,14 +166,22 @@ func addEnv(dir string, cfg *v1alpha1.Environment) error {
...
+		// write spec.json
+		if err := writeJSON(cfg, filepath.Join(path, "spec.json")); err != nil {
+			return err
+		}
 
-	// write main.jsonnet
-	if err := writeJSON(struct{}{}, filepath.Join(path, "main.jsonnet")); err != nil {
-		return err
+		// write main.jsonnet
+		if err := writeJSON(struct{}{}, filepath.Join(path, "main.jsonnet")); err != nil {
```
Comment (jvrplmlmn) [inline cmd/tk/env.go:182]: Since we are writing a jsonnet file, should we use `writeJsonnet` here as well for consistency?

**Review by jvrplmlmn — APPROVED** [review]
👍

File: cmd/tk/util.go [hunk 4]
```diff
@@ -56,3 +58,22 @@ func writeJSON(i interface{}, path string) error {
...
 }
+
+// writeJsonnet writes the given object to the path as a formatted Jsonnet file
+func writeJsonnet(i interface{}, path string) error {
+	out, err := json.MarshalIndent(i, "", "  ")
+	if err != nil {
+		return fmt.Errorf("marshalling: %s", err)
+	}
+
+	main, err := formatter.Format(path, string(out), formatter.DefaultOptions())
```
Comment (sh0rez) [inline cmd/tk/util.go:69]: This should use `tanka.Format` defined in https://github.com/grafana/tanka/blob/17baa356eaea5a30c4a88a8950e2dbefb51636e1/pkg/tanka/format.go#L88-L90

It uses the same functionality underneath, but if we ever might modify the way we format things, we only need to look in one place

**Review by sh0rez — APPROVED** [review]

File: cmd/tk/util.go (hunk 4 above)
Comment (Duologic) [reply cmd/tk/util.go:69]: I noticed, Javier also made a comment on another PR related to writing and formatting files. I hope to consolidate that in another PR.

**Pull request merged**